types/module/configurator.go

```diff
@@ -38,3 +55,54 @@ func (c configurator) MsgServer() grpc.Server {
 func (c configurator) QueryServer() grpc.Server {
 	return c.queryServer
 }
+
+// RegisterMigration implements the Configurator.RegisterMigration method
+func (c configurator) RegisterMigration(moduleName string, forVersion uint64, handler MigrationHandler) error {
+	if forVersion == 0 {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidVersion, "module migration versions should start at 1")
+	}
+
+	if c.migrations[moduleName] == nil {
+		c.migrations[moduleName] = map[uint64]MigrationHandler{}
+	}
+
+	if c.migrations[moduleName][forVersion] != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrLogic, "another migration for module %s and version %d already exists", moduleName, forVersion)
+	}
+
+	c.migrations[moduleName][forVersion] = handler
+
+	return nil
+}
+
+// runModuleMigrations runs all in-place store migrations for one given module from a
+// version to another version.
+func (c configurator) runModuleMigrations(ctx sdk.Context, moduleName string, fromVersion, toVersion uint64) error {
+	// No-op if toVersion is the initial version.
+	// Some modules don't have a store key (e.g. vesting), in this case, their
+	// ConsensusVersion will always stay at 1, and running migrations on
+	// those modules will be skipped on this line.
+	if toVersion <= 1 {
+		return nil
+	}
+
+	moduleMigrationsMap, found := c.migrations[moduleName]
+	if !found {
+		return sdkerrors.Wrapf(sdkerrors.ErrNotFound, "no migrations found for module %s", moduleName)
+	}
+
+	// Run in-place migrations for the module sequentially until toVersion.
+	for i := fromVersion; i < toVersion; i++ {
+		migrateFn, found := moduleMigrationsMap[i]
+		if !found {
+			return sdkerrors.Wrapf(sdkerrors.ErrNotFound, "no migration found for module %s from version %d to version %d", moduleName, i, i+1)
```

Comment:
If a migration is needed for every version even if it's a no-op (which I tend to agree with), we should document this _really_ clearly in the `RegisterMigration` docs, i.e. "you _must_ register a migration for _every_ consensus version increment".

Reply:
I added a godoc comment reflecting this. Will also emphasize it in the ADR and in docs.